node/bft/src/helpers/pending.rs

```diff
@@ -82,33 +82,35 @@ impl<T: Copy + Clone + PartialEq + Eq + Hash, V: Clone> Pending<T, V> {
 
     /// Returns `true` if the pending queue contains the specified `item` for the specified `peer IP`.
     pub fn contains_peer(&self, item: impl Into<T>, peer_ip: SocketAddr) -> bool {
-        self.pending.read().get(&item.into()).map_or(false, |peer_ips| peer_ips.contains(&peer_ip))
+        self.pending.read().get(&item.into()).map_or(false, |peer_ips| peer_ips.contains_key(&peer_ip))
     }
 
     /// Returns the peer IPs for the specified `item`.
-    pub fn get(&self, item: impl Into<T>) -> Option<HashSet<SocketAddr>> {
-        self.pending.read().get(&item.into()).cloned()
+    pub fn get_peers(&self, item: impl Into<T>) -> Option<HashSet<SocketAddr>> {
+        self.pending.read().get(&item.into()).map(|map| map.keys().cloned().collect())
     }
 
     /// Returns the number of pending callbacks for the specified `item`.
     pub fn num_callbacks(&self, item: impl Into<T>) -> usize {
         let item = item.into();
+        let now = OffsetDateTime::now_utc().unix_timestamp();
         // Clear the callbacks that have expired.
-        self.clear_expired_callbacks_for_item(item);
+        self.clear_expired_callbacks_for_item(now, item);
         // Return the number of live callbacks.
-        self.callbacks.lock().get(&item).map_or(0, |callbacks| callbacks.len())
+        self.pending.read().get(&item).map_or(0, |peers| peers.values().flatten().count())
```

Comment:
```suggestion
        self.pending.read().get(&item).map_or(0, |peers| peers.values().map(|v| v.len()).sum::<usize>())
```
I think the flattening would cause the counting to be done in a linear fashion instead of using the readily available lengths

Reply:
Resolved with https://github.com/AleoHQ/snarkOS/pull/3135/commits/a39996d213f0a3e9c2556bb305d1fce2e00bc1d6